src/main/scala/no/gdl/bookapi/service/translation/TranslationService.scala

```diff
@@ -201,6 +166,72 @@ trait TranslationService {
 
       })
     }
+
+    private def verifyAllFilesAreTranslated(inTranslationFiles: Seq[InTranslationFile]): Try[Seq[InTranslationFile]] = {
+      if(inTranslationFiles.forall(_.translationStatus == TranslationStatus.TRANSLATED)) {
+        Success(inTranslationFiles)
+      } else {
+        Failure(new RuntimeException(s"Not all files for translation are finished translating. Cannot continue."))
+      }
+    }
+
+    def findAllTranslationFiles(inTranslationFile: InTranslationFile): Seq[InTranslationFile] = {
+      translationDbService.filesForTranslation(inTranslationFile.inTranslationId)
+    }
+
+    def inTranslationWithId(inTranslationId: Long): Option[InTranslation] = {
+      translationDbService.translationWithId(inTranslationId)
+    }
+
+    private def newTranslationId(inTranslation: InTranslation): Try[Long] = {
+      inTranslation.newTranslationId match {
+        case None => Failure(new RuntimeException(s"The book for ${inTranslation.originalTranslationId} has not yet been translated. Cannot fetch updates."))
+        case Some(newTranslationId) => Success(newTranslationId)
+      }
+    }
+
+    private def originalBook(inTranslation: InTranslation): Try[Book] = {
+      readService.withIdAndLanguage(inTranslation.originalTranslationId, inTranslation.fromLanguage) match {
+        case None => Failure(new RuntimeException(s"The original book with id ${inTranslation.originalTranslationId} was not found. Cannot fetch updates."))
+        case Some(originalBook) => Success(originalBook)
+      }
+    }
+
+    private def existingTranslation(translationId: Long): Try[domain.Translation] = {
+      translationRepository.withId(translationId) match {
+        case None => Failure(new RuntimeException(s"The translated book with id $translationId was not found. Cannot fetch updates."))
+        case Some(existingTranslation) => Success(existingTranslation)
+      }
+    }
+
+    private def findMetadataFile(inTranslation: InTranslation, translationFiles: Seq[InTranslationFile]): Try[InTranslationFile] = {
+      translationFiles.find(_.fileType == FileType.METADATA) match {
+        case None => Failure(new RuntimeException(s"No metadata for translation with id ${inTranslation.id} found. Cannot fetch updates"))
+        case Some(metadataFile) => Success(metadataFile)
+      }
+    }
+
+    private def findChapterFiles(translation: InTranslation, translationFiles: Seq[InTranslationFile]): Try[Seq[InTranslationFile]] = {
+      translationFiles.filter(_.fileType == FileType.CONTENT) match {
+        case head :: tail => Success(head :: tail)
```

Comment:
To do this similarly to the other functions, you could match on `case Nil => Failure(…` first and then you know that the list is nonempty in the other case.